backend/src/test/java/com/redhat/cloud/notifications/events/FromCamelHistoryFillerTest.java

```diff
@@ -0,0 +1,96 @@
+package com.redhat.cloud.notifications.events;
+
+import com.redhat.cloud.notifications.Json;
+import com.redhat.cloud.notifications.MicrometerAssertionHelper;
+import com.redhat.cloud.notifications.TestLifecycleManager;
+import com.redhat.cloud.notifications.db.NotificationResources;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+import io.smallrye.reactive.messaging.connectors.InMemoryConnector;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import javax.enterprise.inject.Any;
+import javax.inject.Inject;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.redhat.cloud.notifications.events.FromCamelHistoryFiller.FROMCAMEL_CHANNEL;
+import static com.redhat.cloud.notifications.events.FromCamelHistoryFiller.MESSAGES_ERROR_COUNTER_NAME;
+import static com.redhat.cloud.notifications.events.FromCamelHistoryFiller.MESSAGES_PROCESSED_COUNTER_NAME;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+public class FromCamelHistoryFillerTest {
+
+    @Inject
+    @Any
+    InMemoryConnector inMemoryConnector;
+
+    @InjectMock
+    NotificationResources notificationResources;
+
+    @Inject
+    MicrometerAssertionHelper micrometerAssertionHelper;
+
+    @BeforeEach
+    void beforeEach() {
+        micrometerAssertionHelper.saveCounterValuesBeforeTest(
+                MESSAGES_PROCESSED_COUNTER_NAME,
+                MESSAGES_ERROR_COUNTER_NAME
+        );
+    }
+
+    @AfterEach
+    void afterEach() {
+        micrometerAssertionHelper.clearSavedValues();
+    }
+
+    @Test
+    void testInvalidPayload() {
+        inMemoryConnector.source(FROMCAMEL_CHANNEL).send("I am not valid!");
+
+        micrometerAssertionHelper.awaitAndAssertCounterIncrement(MESSAGES_PROCESSED_COUNTER_NAME, 1);
+        micrometerAssertionHelper.assertCounterIncrement(MESSAGES_ERROR_COUNTER_NAME, 1);
+
+        verifyNoInteractions(notificationResources);
+    }
+
+    @Test
+    void testValidPayload() {
+        String expectedHistoryId = UUID.randomUUID().toString();
+        String expectedOutcome = "Success";
+        int expectedDuration = 123;
+
+        String payload = Json.encode(Map.of(
+                "id", expectedHistoryId,
+                "data", Json.encode(Map.of(
+                        "outcome", expectedOutcome,
+                        "details", Collections.emptyMap(),
```

Comment:
Do you have an example of data that field could contain?

Reply:
`2021-12-14 11:08:24,252 INFO  [com.red.clo.not.eve.FromCamelHistoryFiller] (vert.x-eventloop-thread-5) Processing return from camel: {"specversion":"1.0","source":"demo-log","type":"com.redhat.cloud.notifications.history","time":"2021-12-14T10:08:23.217Z","id":"e3c90a94-751b-4ce1-b345-b85d825795a4","content-type":"application\/json","data":"{\"duration\":67549274,\"finishTime\":1639476503209,\"details\":{\"type\":\"com.redhat.console.notification.toCamel.tower\",\"target\":\"1.2.3.4\"},\"outcome\":\"com.jayway.jsonpath.PathNotFoundException: Missing property in path $['bla']\",\"successful\":false}"}
`

Reply:
Thanks